### PR TITLE
feat: Allow custom express handler after deployd middleware

### PR DESCRIPTION
**Pull request by evaisse**
I would like to let express or node http server to handle 404 & eventual redirects AFTER deployd routing consider unable to handle the request.

## Description

We look into the first `handleRequest`call to check if their is a next() handler function. If it as one, we consider that the app is part of an custom deployment script or an express one. We do pass the native handler to deployd router to override deployd default 404 handler.

## Motivation and Context

I would redirecting all non handled request to my own express logic afterward the request had been pass through deployd routing. On some other projects make custom 404 pages this way.

For exemple, attaching an express handler AFTER the deployd middleware : 

```js
// test deployd routing pass to following handler 
// when a given page match an express handler
app.get('*', function(req, res) {
	res.sendFile('public/index.html');
});
```
But we can see many more examples.

## How Has This Been Tested?

Don't exactly how test it, I do have updated the `test-attach.js` script but I don't see any test running it...

## Types of changes

- [X] Breaking change 

Possible breaking changes if we consider changing the final handling behavior of deployd routing.

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by andreialecu** [comment]
Tests seem to be failing, not sure if it is related to this PR or some other issue. Did you get tests to run locally?

**Comment by evaisse** [comment]
Using a custom docker env, I did manage to run the tests : 

```
docker run --name some-mongo -p 27017:27017 mongo:3.2
```

But still i have some troubles whit one test, but I'm  pretty sure it's related to my changes.

```

  Context
    .url
      ✓ should not have the base url
    .done
      ✓ should provide default headers
      1) should provide default headers

  56 passing (2s)
  1 failing

  1) Context
       .done
         should provide default headers:
     Uncaught TypeError: Cannot read property 'apply' of undefined
      at Request._callback (test/support.js:43:16)
      at Request.self.callback (node_modules/request/request.js:187:22)
      at Request.<anonymous> (node_modules/request/request.js:1044:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:965:12)
      at endReadableNT (_stream_readable.js:1064:12)
      at _combinedTickCallback (internal/process/next_tick.js:138:11)
      at process._tickDomainCallback (internal/process/next_tick.js:218:9)


```

**Comment by christianguevara** [comment]
> Using a custom docker env, I did manage to run the tests :
> 
> ```
> docker run --name some-mongo -p 27017:27017 mongo:3.2
> ```
> But still i have some troubles whit one test, but I'm pretty sure it's related to my changes.
> 
> ```
> 
>   Context
>     .url
>       ✓ should not have the base url
>     .done
>       ✓ should provide default headers
>       1) should provide default headers
> 
>   56 passing (2s)
>   1 failing
> 
>   1) Context
>        .done
>          should provide default headers:
>      Uncaught TypeError: Cannot read property 'apply' of undefined
>       at Request._callback (test/support.js:43:16)
>       at Request.self.callback (node_modules/request/request.js:187:22)
>       at Request.<anonymous> (node_modules/request/request.js:1044:10)
>       at IncomingMessage.<anonymous> (node_modules/request/request.js:965:12)
>       at endReadableNT (_stream_readable.js:1064:12)
>       at _combinedTickCallback (internal/process/next_tick.js:138:11)
>       at process._tickDomainCallback (internal/process/next_tick.js:218:9)
> ```

Is not related, forked the repo, installed deps and run 'npm run test' and the same error on the same lines.

**Comment by andreialecu** [comment]
@christianguevara a fix for those problems was merged. There are still some problems with this PR though.

**Comment by evaisse** [comment]
I did just fix the last tests, can you give another review ?

**Review by andreialecu — APPROVED** [review]
LGTM

File: lib/resources/files.js [hunk 1]
```diff
@@ -31,8 +31,7 @@ Files.prototype.handle = function (ctx, next) {
 
   send(ctx.req, url.parse(ctx.url).pathname, {root: path.resolve(this.public)})
     .on('error', function (err) {
-      ctx.res.statusCode = 404;
-      respond('Resource Not Found', ctx.req, ctx.res);
+      return next();
```
[the remark this answers is not in the document]
Comment (evaisse) [reply lib/resources/files.js]: We do return a 404. this handler is only run after no resources can be found.

**Pull request merged**